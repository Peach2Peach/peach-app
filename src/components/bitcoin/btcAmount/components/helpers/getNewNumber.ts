import { SATSINBTC } from '../../../../../constants'

export const getNewNumber = (value: number) => {
  const newNumber = (value / SATSINBTC).toFixed(8).split('')
  for (let i = newNumber.length - 3; i > 0; i -= 3) {
    if (newNumber[i] !== '.') {
      newNumber.splice(i, 0, ' ')
    }
  }
  return newNumber
}
