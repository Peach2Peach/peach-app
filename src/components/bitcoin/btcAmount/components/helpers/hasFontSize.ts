import { TextStyle } from 'react-native'

export const hasFontSize = (s: false | TextStyle): s is TextStyle => s !== false && s?.fontSize !== undefined
