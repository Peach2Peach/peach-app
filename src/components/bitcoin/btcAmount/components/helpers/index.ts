export { getLetterSpacing } from './getLetterSpacing'
export { getNewNumber } from './getNewNumber'
