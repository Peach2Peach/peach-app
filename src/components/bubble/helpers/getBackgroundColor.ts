import tw from '../../../styles/tailwind'
import { BubbleProps } from '../Bubble'

export const getBackgroundColor = ({ color, ghost }: Pick<BubbleProps, 'color' | 'ghost'>) => {
  if (color === 'primary') return ghost ? tw`bg-primary-background-light` : tw`bg-primary-main`
  if (color === 'primary-mild') return ghost ? tw`bg-primary-background-light` : tw`bg-primary-background-dark`

  return tw`transparent`
}
