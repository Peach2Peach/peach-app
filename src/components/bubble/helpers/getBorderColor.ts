import tw from '../../../styles/tailwind'
import { BubbleProps } from '../Bubble'

export const getBorderColor = ({ color, ghost }: Pick<BubbleProps, 'color' | 'ghost'>) => {
  if (color === 'primary') return ghost ? tw`border-primary-main` : undefined
  if (color === 'primary-mild') return ghost ? tw`border-black-100` : tw`border-primary-mild-1`
  if (color === 'gray') return tw`border-black-50`

  return tw`border-black-100`
}
