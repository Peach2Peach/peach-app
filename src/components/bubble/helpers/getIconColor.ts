import tw from '../../../styles/tailwind'
import { BubbleProps } from '../Bubble'
import { getTextColor } from './getTextColor'

export const getIconColor = ({ color, ghost }: Pick<BubbleProps, 'color' | 'ghost'>) => {
  if (color === 'primary-mild') return ghost ? tw`text-black-1` : tw`text-primary-main`
  return getTextColor({ color, ghost })
}
