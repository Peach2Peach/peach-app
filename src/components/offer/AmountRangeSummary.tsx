import { View } from 'react-native'
import tw from '../../styles/tailwind'
import i18n from '../../utils/i18n'
import { SatsFormat, Text } from '../text'
import { TradeSeparator } from './TradeSeparator'

type Props = {
  amount: number[]
}
export const AmountRangeSummary = ({ amount }: Props) => (
  <View style={tw`gap-2px`}>
    <TradeSeparator text={i18n('amount')} />
    <View>
      <SatsFormat
        sats={amount[0]}
        style={[tw`text-lg font-semibold subtitle-1`, tw.md`text-xl`]}
        satsStyle={[tw`font-normal text-3xs`, tw.md`text-base`]}
        bitcoinLogoStyle={[tw`w-3 h-3 mr-1 -mt-1`, tw.md`w-4 h-4`]}
      />
      <Text style={tw`-mt-2 -mb-1 button-small text-black-2`}>{i18n('upTo')}</Text>
      <SatsFormat
        sats={amount[1]}
        style={[tw`text-lg font-semibold subtitle-1`, tw.md`text-xl`]}
        satsStyle={[tw`font-normal text-3xs`, tw.md`text-base`]}
        bitcoinLogoStyle={[tw`w-3 h-3 mr-1 -mt-1`, tw.md`w-4 h-4`]}
      />
    </View>
  </View>
)
