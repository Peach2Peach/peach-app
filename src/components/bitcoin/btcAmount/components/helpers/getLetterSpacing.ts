import { TextStyle } from 'react-native'
import { hasFontSize } from './hasFontSize'

export const getLetterSpacing = (style: (false | TextStyle)[]) => {
  const styleWithFontSize = style.find(hasFontSize)
  const fontSize = styleWithFontSize?.fontSize ?? 22

  return {
    whiteSpace: -(fontSize * 0.35),
    dot: -(fontSize * 0.21),
    digit: -(fontSize * 0.075),
  }
}
