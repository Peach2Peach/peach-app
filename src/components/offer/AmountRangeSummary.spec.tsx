import { createRenderer } from 'react-test-renderer/shallow'
import { buyOffer } from '../../../tests/unit/data/offerData'
import { AmountRangeSummary } from './AmountRangeSummary'
import { mockDimensions } from '../../../tests/unit/helpers/mockDimensions'

describe('AmountRangeSummary', () => {
  const renderer = createRenderer()
  it('renders correctly', () => {
    renderer.render(<AmountRangeSummary amount={buyOffer.amount} />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
  it('renders correctly for md screens', () => {
    mockDimensions({ width: 600, height: 840 })
    renderer.render(<AmountRangeSummary amount={buyOffer.amount} />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
})
