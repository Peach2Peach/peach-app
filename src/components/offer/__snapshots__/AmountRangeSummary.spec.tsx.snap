// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountRangeSummary renders correctly 1`] = `
<View
  style={
    {
      "gap": 2,
    }
  }
>
  <Divider
    text="amount"
  />
  <View>
    <SatsFormat
      bitcoinLogoStyle={
        [
          {
            "height": 12,
            "marginRight": 4,
            "marginTop": -4,
            "width": 12,
          },
          {
            "height": 16,
            "width": 16,
          },
        ]
      }
      sats={50000}
      satsStyle={
        [
          {
            "fontSize": 12,
            "fontWeight": "normal",
          },
          {
            "fontSize": 16,
          },
        ]
      }
      style={
        [
          {
            "fontFamily": "Baloo2-SemiBold",
            "fontSize": 16,
            "fontWeight": "600",
            "letterSpacing": 0.16,
            "lineHeight": 28,
          },
          {
            "fontSize": 20,
          },
        ]
      }
    />
    <PeachText
      style={
        {
          "color": "#7D675E",
          "fontFamily": "Baloo2-SemiBold",
          "fontSize": 13,
          "letterSpacing": 0.47,
          "lineHeight": 22,
          "marginBottom": -4,
          "marginTop": -8,
          "textTransform": "uppercase",
        }
      }
    >
      up to
    </PeachText>
    <SatsFormat
      bitcoinLogoStyle={
        [
          {
            "height": 12,
            "marginRight": 4,
            "marginTop": -4,
            "width": 12,
          },
          {
            "height": 16,
            "width": 16,
          },
        ]
      }
      sats={250000}
      satsStyle={
        [
          {
            "fontSize": 12,
            "fontWeight": "normal",
          },
          {
            "fontSize": 16,
          },
        ]
      }
      style={
        [
          {
            "fontFamily": "Baloo2-SemiBold",
            "fontSize": 16,
            "fontWeight": "600",
            "letterSpacing": 0.16,
            "lineHeight": 28,
          },
          {
            "fontSize": 20,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`AmountRangeSummary renders correctly for md screens 1`] = `
<View
  style={
    {
      "gap": 2,
    }
  }
>
  <Divider
    text="amount"
  />
  <View>
    <SatsFormat
      bitcoinLogoStyle={
        [
          {
            "height": 12,
            "marginRight": 4,
            "marginTop": -4,
            "width": 12,
          },
          {
            "height": 16,
            "width": 16,
          },
        ]
      }
      sats={50000}
      satsStyle={
        [
          {
            "fontSize": 12,
            "fontWeight": "normal",
          },
          {
            "fontSize": 16,
          },
        ]
      }
      style={
        [
          {
            "fontFamily": "Baloo2-SemiBold",
            "fontSize": 16,
            "fontWeight": "600",
            "letterSpacing": 0.16,
            "lineHeight": 28,
          },
          {
            "fontSize": 20,
          },
        ]
      }
    />
    <PeachText
      style={
        {
          "color": "#7D675E",
          "fontFamily": "Baloo2-SemiBold",
          "fontSize": 13,
          "letterSpacing": 0.47,
          "lineHeight": 22,
          "marginBottom": -4,
          "marginTop": -8,
          "textTransform": "uppercase",
        }
      }
    >
      up to
    </PeachText>
    <SatsFormat
      bitcoinLogoStyle={
        [
          {
            "height": 12,
            "marginRight": 4,
            "marginTop": -4,
            "width": 12,
          },
          {
            "height": 16,
            "width": 16,
          },
        ]
      }
      sats={250000}
      satsStyle={
        [
          {
            "fontSize": 12,
            "fontWeight": "normal",
          },
          {
            "fontSize": 16,
          },
        ]
      }
      style={
        [
          {
            "fontFamily": "Baloo2-SemiBold",
            "fontSize": 16,
            "fontWeight": "600",
            "letterSpacing": 0.16,
            "lineHeight": 28,
          },
          {
            "fontSize": 20,
          },
        ]
      }
    />
  </View>
</View>
`;
